libdnf/conf/option_string.cpp

```diff
@@ -80,7 +82,7 @@ void OptionString::set(Priority priority, const std::string & value) {
 
 const std::string & OptionString::get_value() const {
     if (get_priority() == Priority::EMPTY) {
-        throw ValueNotSet();
+        throw OptionValueNotSetError(M_("Option value is not set"));
```

Comment:
But the option does not know its name. The caller of the `get_value` method knows which option it is working with. It can handle the exception. E.g. wrap the message and add name of the option.
Or do we want to store another redundant string with option name for each option?

Including the section ... Which section? Option knows nothing about sections. The option has type, value, default value, priority and certain rules. But it doesn't know anything about *.ini or other sections. The option can be used anywhere. In config, but also in the argument parser.

Reply:
Yes, I know. I don't have a good solution to this, but we may have to add the name to the options. We're calling this from a ton of places and it wouldn't make sense to wrap all those with a try-catch and the same copy-pasted rethrow code.

Similar with the section, from a user PoV, you should get the full "path" of the option in question. But that'd be even worse to implement and we really only have `[main]` and repos, so we may skimp on this.